dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/ArtifactProblem.java

```diff
@@ -78,4 +81,44 @@ public static String formatProblems(Iterable<ArtifactProblem> problems) {
     }
     return output.toString();
   }
+
+  @Override
+  public boolean equals(Object other) {
```

Comment:
Equality function is added to dedup at https://github.com/GoogleCloudPlatform/cloud-opensource-java/pull/1210/files#diff-659d7bf4148f942fcc1eb60a50a2ee80R261

Reply:
Equality test is added in ArtifactProblemTest.